financial-templates-lib/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -0,0 +1,146 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+
+// An implementation of PriceFeedInterface that uses CryptoWatch to retrieve prices.
+class CryptoWatchPriceFeed extends PriceFeedInterface {
+  // Constructs the CryptoWatchPriceFeed.
+  // apiKey the CW API key. Note: these API keys are rate-limited.
+  // exchange a string identifier for the echange to pull prices from. This should be the identifier used to
+  //          identify the exchange in CW's REST API.
+  // pair a string representation of the pair the price feed is tracking. This pair should be available on the
+  //      provided exchange. The string should be the representation used by CW to identify this pair.
+  // lookback how far in the past the historical prices will be available using the getHistoricalPrice function.
+  // networker networker object used to send the API requests.
+  // getTime function to return the current time.
+  // minTimeBetweenUpdates min number of seconds between updates. If update() is called again before this number of
+  // seconds has passed, it will be a no-op.
+  constructor(web3, apiKey, exchange, pair, lookback, networker, getTime, minTimeBetweenUpdates) {
```

Comment:
This looks great but should we not ingest a logger like with our other infrastructure to emit events on updates and throw errors when something breaks?

Reply:
Done. PTAL.